### DISCUSSION TITLE
[TOPAS validation]: NumberOfSequentialTimes problem for small voxels

### DISCUSSION THREAD

**Discussion by claus-e-andersen** [OpenTOPAS/OpenTOPAS#65] [Validation]
### Contact Details

clan@dtu.dk

### Describe your simulation result

The results obtained with NumberOfSequentialTimes > 1 can be seriously off for small voxels. I demonstrate this with a simple example where the same simulation is repeated with different settings of  NumberOfSequentialTimes while keeping the total number of histories at 1E7.

------------------------------
Topas:  NumberOfSequentialTimes problem for small voxel sizes
I often use the product of the two parameters (i) NumberOfSequentialTimes and (ii) NumberOfHistoriesInRun to specify the total number of histories in my simulations. For example, this is an easy way to get above 1e9 histories in a single simulation. However, for the reasons given below, if think there is a problem with this approach and I’ll now only use NumberOfSequentialTimes = 1. There seems to be a serious bug in the TimeFeature (TF) system or I'm doing something wrong.  

System
openTopas version 4.0 on a linux PC running Ubuntu 22.04.

Demonstration
I’m shooting 10 MeV electrons on a box of water. In the center of the box, I have a rod in the z-direction. Here I score the dose to the medium with zbin set to 300, as to give me an electron beam depth-dose curve with a resolution of 0.2 mm. I do four simulations (A to D), all with 1E7 total number of histories realized as follows:
(A)	NumberOfSequentialTimes = 1 NumberOfHistoriesInRun= 1E7
(B)	NumberOfSequentialTimes = 10 NumberOfHistoriesInRun= 1E6
(C)	NumberOfSequentialTimes = 100 NumberOfHistoriesInRun= 1E5
(D)	NumberOfSequentialTimes = 1000 NumberOfHistoriesInRun= 1E4

The results are shown in Figure 1 for a version of the job file where HLX and HLY for the center rod were both set to 5 mm. Figure 1 shows that the four computations give essentially identical results independently on how I realize the 1e7 total number of histories. So, all is good so far.

However, I then set HLX and HLY for the center rod to 0.2 mm, and I get the results in Figure 2. Now there are huge differences between the four simulations. The results with NumberOfSequentialTimes different from 1 are seriously off. 

Figure 3 presents the same data as Figure 2 but re-scaled such that it is now clear that all results are very similar in structure and variation (except for some scaling parameter).

I get the same results also if I explicitly set the TimelineEnd to 10 s.

I score both the mean dose and the sum dose. The ratio between the two is always 1E7 as sxpected.

I attach the figures and the simple parameter file where HLX and HLY are set to 0.2 mm. 

Conclusion
There seems to be a serious bug in the openTopas v. 4 related to the statistics procedure used for aggregating results with the time feature (TF) system or am I doing something wrong? 
For now I’ll only use NumberOfSequentialTimes = 1.  

With best regards
Claus

Claus E. Andersen
DTU Health Tech
Denmark

![Figure-1-NumberOfSequentialTimes-HLX5mm](https://github.com/user-attachments/assets/3047209f-e17c-4153-b48d-947fab1deb02)
![Figure-2-NumberOfSequentialTimes-HLX0p2mm](https://github.com/user-attachments/assets/c3b07cf6-b5da-4524-baa3-db832551da11)
![Figure-3-NumberOfSequentialTimes-HLX0p2mm](https://github.com/user-attachments/assets/94c06431-1615-435f-90b1-4c1a61092d4a)
[NOST-openTopas-4p0-output-1000.csv](https://github.com/user-attachments/files/16622449/NOST-openTopas-4p0-output-1000.csv)
[Testing-NumberOfSequentialTimes-job.txt](https://github.com/user-attachments/files/16622451/Testing-NumberOfSequentialTimes-job.txt)

### Version

4.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JoseRamosMendez** [comment]
Thanks Claus for reporting this unfortunate behavior. Indeed, the issue is triggered only when the mean is also requested in the scorer. Otherwise, if only the sum (default) is called, the behavior is normal as expected. We'll work in a fix for this. Best!

**Comment by claus-e-andersen** [comment]
Thanks a lot. I ran the tests again, and I can confirm that the sum-only scorer indeed works perfectly. I also ran the test with both sum AND standard_deviation in the scorer report, and that also worked. So, "mean" is the bad guy.  Thanks again. - Claus

**Reply by JoseRamosMendez** [reply]
Thanks for the extra test!
